pkgs/by-name/gl/globalplatform/package.nix

```diff
@@ -0,0 +1,87 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pcsclite,
+  PCSC,
+  pkg-config,
+  cmake,
+  zlib,
+  pandoc,
+  doxygen,
+  graphviz,
+  openssl,
+  cmocka,
+  makeWrapper,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "globalplatform";
+  version = "2.4.0-unreleased";
+
+  src = fetchFromGitHub {
+    owner = "kaoh";
+    repo = "globalplatform";
+    rev = "c901eabd771379e43ebd71bcf8071398b868a591";
+    sha256 = "sha256-dGbGDDlQ01crqS9J1cTd9xNpagKgdV+ha4HtnPK8SZo=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    pandoc
+    doxygen
+    graphviz
+    makeWrapper
+  ];
+
+  buildInputs =
+    [
+      zlib
+      openssl
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      pcsclite
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      PCSC
+    ];
+
+  cmakeFlags = [
+    "-DTESTING=ON"
+  ];
+
+  doCheck = true;
+
+  nativeCheckInputs = [
+    cmocka
+  ];
+
+  preCheck = ''
+    cp "$src/gpshell/helloworld.cap" globalplatform/src
+    cp "$src/globalplatform/src/rsa_pub_key_test.pem" globalplatform/src
+  '';
+
+  # gpshell uses dlopen() to load libgppcscconnectionplugin.so at runtime
+
+  wrapperArgs =
+    lib.optionals stdenv.hostPlatform.isLinux [
+      "--prefix LD_LIBRARY_PATH : ${placeholder "out"}/lib"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      "--prefix DYLD_LIBRARY_PATH : ${placeholder "out"}/lib"
+    ];
+
+  preFixup = ''
+    wrapProgram $out/bin/gpshell $wrapperArgs
+  '';
```

Comment:
Have you tried using patchelf to add the library to the rpath of the executable directly? I think avoiding wrapping is usually preferrable, if possible.

Reply:
Yes I tried that, and it did now work because dlopen() seemed to not respect RPATH, but I will try again.

Reply:
Okay, I managed to fix it - turns out I needed to set the RPATH of the library which was implementing the dlopen() call, no the application binary which was using this intermediate library.